inference-engine/src/mkldnn_plugin/nodes/mkldnn_eltwise_node.cpp

```diff
@@ -140,11 +140,29 @@ struct jit_uni_eltwise_generic : public MKLDNNPlugin::jit_uni_eltwise_kernel, pu
 
         this->preamble();
 
-        for (int i = 0; i < jep.inputs_number; i++)
-            mov(get_src_reg(i), ptr[reg_params + GET_OFF(src_ptr[0]) + i * sizeof(size_t)]);
-        mov(reg_dst, ptr[reg_params + GET_OFF(dst)]);
-        mov(reg_work_amount, ptr[reg_params + GET_OFF(work_amount)]);
-        mov(reg_oc_off, ptr[reg_params + GET_OFF(oc_off)]);
+        const int offset_count = jep.input_size - 1;
+
+        // ptrs initializing
+        auto init_ptrs_with_offsets = [this, offset_count](Reg64 pointer, const std::vector<size_t>& offsets) {
+            for (int j = 0; j < offset_count; j++) {
+                mov(reg_tmp_64, offsets[j]);
+                imul(reg_tmp_64, ptr[reg_indexes + j * sizeof(size_t)]);
+                add(pointer, reg_tmp_64);
```

Comment:
In fact the routine can be further optimized:
1) You can skip "mov" and "imul" operations in case offsets[j] equals zero.
2) You also can pass dims as JIT kernel static params. In that case if dims[j] == 1 it is not necessary to add offset at all (this is very often case actually considering broadcasting support)

Reply:
Added, thank you!